### PR TITLE
Endpoints products and stock

### PR DESCRIPTION
**Pull request by AngheloZambrana**
Testing steps
1. Remove docker containers
2. docker compose up --build
3. http://localhost:8000/products/index.html
4. Go to Stock section

**First endpoint GetStocksBySubsidiaryAndProductCode**
This is /Stock/subsidiary/{subsidiaryId}/product-code/{productCode}
subsidiaryID: acc847fa-7327-497e-b89f-1e963518f401
ProductCode: 40

This should show you the information of the specific product in that subsidiary.

When entering this data in the endpoint it should show the stock information. 

**Second endpoint UpdateStocks** 
This is /Stock/update-multiple

1. Execute a GetAll and check the first two stocks found.
2. When you press run enter this to update the first 2 stocks
[
 {
  “stockId": ‘00bf68b6-81c4-47ba-9aa8-9bdcbfd81686’,
  “code": 20,
  “quantity": 3,
  “productId": ‘ca53a876-26ae-43d2-b5f0-81e6d96d9e3f’,
  “subsidiaryId": ”87c28dd0-c12b-482a-b0e2-548a7021668b”
 },
 {
  “stockId": ‘0121330a-2be7-4c18-b853-6999229fa2ff’,
  }, { “code”: 5,
  }, “quantity”: 10,
  “productId": ‘7bfccdc1-41bf-4175-bdbb-9ad8a7e2ce48’,
  “subsidiaryId": ”111dd672-c6fe-4eeb-b224-6005f446ca39”
 }
]
PLEASE FOLLOW THE STRCTURE OF THE PICTURE
image

3. rerun the GetAll and verify that the data was updated.

### CODE REVIEW

**Review by DeidamiaFuentes — APPROVED** [review]

File: apps/ProductManagement/Backend/Services/StockService.cs [hunk 1]
```diff
@@ -176,4 +176,51 @@ public StockDTO UpdateStock(StockWithoutIDDTO stock, Guid stockId)
...
+                updatedStocks.Add(stockDTO);
+            }
+        }
+        return updatedStocks;
+    }
+    
+    public StockDTO? GetStocksBySubsidiaryAndProductCode(Guid subsidiaryId, int productCode)
+    {
+        var stock = _stockDao.ReadAll().FirstOrDefault(s => s.SubsidiaryId == subsidiaryId && s.Code == productCode);
+        var product = _productDao.Read(stock.ProductId);
```
Comment (OropezaHugo) [inline apps/ProductManagement/Backend/Services/StockService.cs]: the endpoint must get the coincidence with the product code, no the stock code

File: apps/ProductManagement/Backend/Services/StockService.cs [hunk 2]
```diff
@@ -176,4 +176,51 @@ public StockDTO UpdateStock(StockWithoutIDDTO stock, Guid stockId)
...
+        List<Category> categories = new List<Category>();
+        if (stock != null)
```
Comment (OropezaHugo) [inline apps/ProductManagement/Backend/Services/StockService.cs]: this validation should be executed before using it to search the product with its content

**Review by OropezaHugo — CHANGES_REQUESTED** [review]

File: apps/ProductManagement/Backend/Services/StockService.cs (hunk 1 above)
Comment (AngheloZambrana) [reply apps/ProductManagement/Backend/Services/StockService.cs]: its fix can you review

File: apps/ProductManagement/Backend/Services/StockService.cs (hunk 2 above)
Comment (AngheloZambrana) [reply apps/ProductManagement/Backend/Services/StockService.cs]: interesting comment I see that for my new fixes

**Review by OropezaHugo — APPROVED** [review]

**Review by JCarlosHidalgoSosa — APPROVED** [review]

**Pull request merged**